### PR TITLE
User reservation data display implemented

### PR DESCRIPTION
**Pull request by MaskaSaw**

### CODE REVIEW

File: hotels-backend/Hotels/Controllers/UsersController.cs [hunk 1]
```diff
@@ -85,6 +85,64 @@ public async Task<ActionResult<IEnumerable<Reservation>>> GetReservations(int id
             return Forbid();
         }
 
+        [Authorize]
+        [HttpGet("{id}/Reservations/Detailed")]
+        public async Task<ActionResult<IEnumerable<ReservationDTO>>> GetReservationsDetailed(int id, [FromQuery] bool all)
+        {
+            var hotels = await _context.Hotels
```
Comment (andrei-kruhliakou-itechart) [inline hotels-backend/Hotels/Controllers/UsersController.cs]: Querying logic in this method should be simplified. You should be able to get all information with only one query starting from Reservation.

File: hotels-client-frontend/src/app/authentication/auth.service.ts [hunk 2]
```diff
@@ -11,6 +11,7 @@ import { environment } from '../../environments/environment';
 import { ApiPaths } from '../api-paths';
 
 @Injectable({ providedIn: 'root'})
+
```
Comment (andrei-kruhliakou-itechart) [inline hotels-client-frontend/src/app/authentication/auth.service.ts]: Why did you add empty line here?

File: hotels-client-frontend/src/styles.less [hunk 3]
```diff
@@ -169,6 +169,43 @@ ul {
...
+        border-radius: 5px;
+
+        tbody {
+            margin-top: 20px;
+        }
+        
+        td {
+            border-bottom: 1px solid @customDarkBlue;
+        }
+
```
Comment (andrei-kruhliakou-itechart) [inline hotels-client-frontend/src/styles.less]: Unnecessary empty line

**Review by andrei-kruhliakou-itechart — CHANGES_REQUESTED** [review]

File: hotels-backend/Hotels/Controllers/UsersController.cs [hunk 4]
```diff
@@ -85,6 +85,50 @@ public async Task<ActionResult<IEnumerable<Reservation>>> GetReservations(int id
...
+                        ArrivalTime = temp.reservation.ArrivalTime,
+                        DepartureTime = temp.reservation.DepartureTime,
+                        StartDate = temp.reservation.StartDate,
+                        EndDate = temp.reservation.EndDate,
+                        ReservationServices = temp.reservation.ReservationServices,
+                        RoomNumber = temp.room.RoomNumber,
+                        HotelName = hotel.Name,
+                        Country = hotel.Country,
+                        City = hotel.City
+                    })
```
Comment (andrei-kruhliakou-itechart) [inline hotels-backend/Hotels/Controllers/UsersController.cs]: You can avoid these joins if you use navigation properties on Reservation entity instead.

**Review by andrei-kruhliakou-itechart — CHANGES_REQUESTED** [review]

File: hotels-backend/Hotels/Startup.cs [hunk 5]
```diff
@@ -35,7 +35,9 @@ public Startup(IConfiguration configuration)
         public void ConfigureServices(IServiceCollection services)
         {
             services.AddDbContext<HotelsDBContext>(opt => opt.UseSqlServer(Configuration["ConnectionStrings:DefaultConnection"]));
-            services.AddControllers();
+            services.AddControllers().AddNewtonsoftJson(options =>
+                options.SerializerSettings.ReferenceLoopHandling = Newtonsoft.Json.ReferenceLoopHandling.Ignore
+                    );
```
Comment (andrei-kruhliakou-itechart) [inline hotels-backend/Hotels/Startup.cs]: Unnecessary indent

File: hotels-backend/Hotels/Startup.cs [hunk 6]
```diff
@@ -35,7 +35,9 @@ public Startup(IConfiguration configuration)
         public void ConfigureServices(IServiceCollection services)
         {
             services.AddDbContext<HotelsDBContext>(opt => opt.UseSqlServer(Configuration["ConnectionStrings:DefaultConnection"]));
-            services.AddControllers();
+            services.AddControllers().AddNewtonsoftJson(options =>
+                options.SerializerSettings.ReferenceLoopHandling = Newtonsoft.Json.ReferenceLoopHandling.Ignore
```
Comment (andrei-kruhliakou-itechart) [inline hotels-backend/Hotels/Startup.cs]: Why do you need this?

Comment (MaskaSaw) [reply hotels-backend/Hotels/Startup.cs]: Without it i get looping error while quering hotels from database

Comment (andrei-kruhliakou-itechart) [reply hotels-backend/Hotels/Startup.cs]: And how exactly is JSON serializer related to database queries?

File: hotels-backend/Hotels/Controllers/HotelsController.cs [hunk 7]
```diff
@@ -71,15 +72,27 @@ public HotelsController(HotelsDBContext context, ImageService imageService)
...
+                    {
+                        Id = hotel.Id,
+                        Name = hotel.Name,
+                        Country = hotel.Country,
+                        City = hotel.City,
+                        Address = hotel.Address,
+                        Image = hotel.Image,
+                        Services = hotel.Services,
+                        RoomCount = hotel.Rooms.Count
+                    })
```
Comment (andrei-kruhliakou-itechart) [inline hotels-backend/Hotels/Controllers/HotelsController.cs]:
```suggestion
                .Select(hotel => 
                    new HotelDTO
                    {
                        Id = hotel.Id,
                        Name = hotel.Name,
                        Country = hotel.Country,
                        City = hotel.City,
                        Address = hotel.Address,
                        Image = hotel.Image,
                        Services = hotel.Services,
                        RoomCount = hotel.Rooms.Count
                    }
                )
```
When creating new objects using multiline object initializer inside lambdas or method calls it is preferable to place new operator on a new line and indent {} accordingly so you can clearly see what belongs to object constructor.

For multiline method calls placing closing parenthesis on a new line can again clearly show you the boundaries of this method call.

**Review by andrei-kruhliakou-itechart — CHANGES_REQUESTED** [review]

File: hotels-backend/Hotels/Controllers/HotelsController.cs (hunk 7 above)
Comment (MaskaSaw) [reply hotels-backend/Hotels/Controllers/HotelsController.cs]: Understood

**Review by andrei-kruhliakou-itechart — APPROVED** [review]

**Pull request merged**